src/app/components/mnoe-users-local-list/mnoe-users-local-list.coffee

```diff
@@ -78,6 +78,66 @@
           MnoErrorsHandler.processServerError(error)
       ).finally(-> user.isSendingInvite = false)
 
+    scope.availableRoles = () ->
+      [
+        "mnoe_admin_panel.constants.user_roles.member",
+        "mnoe_admin_panel.constants.user_roles.admin",
+        "mnoe_admin_panel.constants.user_roles.super_admin"
+      ]
+
+    scope.indexOfRoleInTable = (role) ->
+      scope.availableRoles().indexOf(scope.tlKeyFromUserRole(role))
+
+    # role here should always be in English (as returned by MnoHub)
+    scope.tlKeyFromUserRole = (role) ->
+      enTranslationTable = $translate.getTranslationTable('en-AU')
+      _.findKey(enTranslationTable, _.partial(_.isEqual, role))
+
+    scope.updateUserMail = (user) ->
+      user.isUpdatingEmail = true
+      MnoeUsers.updateStaff(user).then(
+        (response) ->
+          toastr.success('mnoe_admin_panel.dashboard.users.widget.local_list.email_update_sent', {extraData: {email: user.email}})
+        (error) ->
+          toastr.error('mnoe_admin_panel.dashboard.users.widget.local_list.email_update_error')
+          MnoErrorsHandler.processServerError(error)
+      ).finally(-> user.isUpdatingEmail = false)
+
+    scope.updateUserRole = (user) ->
+      user.isUpdatingRole = true
+      # We need to send to MnoHub the role value in English
+      $translate(user.role).then((tls) ->
+        user.role = tls
+        MnoeUsers.updateMember(scope.organization, user).then(
+          (response) ->
+            toastr.success('mnoe_admin_panel.dashboard.users.widget.local_list.role_update_success', {extraData: {user: "#{user.name} #{user.surname}", role: user.role}})
+            user.role = scope.tlKeyFromUserRole(user.role)
```

Comment:
I think this should happen before toastr, so if toastr failed at least user role would be updated in the dom